content/300-guides/050-database/500-using-prisma-with-planetscale.mdx

```diff
@@ -0,0 +1,181 @@
+
+## Commonalities with other database providers
+
+Many aspects of using Prisma with PlanetScale are just like using Prisma with any other relational database. You can still:
+
+- model your database with the [Prisma Schema Language](/concepts/components/prisma-schema)
+- use Prisma's existing [`mysql` database connector](/concepts/database-connectors/mysql) in your schema, along with the [connection string PlanetScale provides you](https://docs.planetscale.com/concepts/connection-strings)
+- use [Introspection](/concepts/components/introspection) for existing projects if you already have a database schema in PlanetScale
+- use [`db push`](/concepts/components/prisma-migrate/db-push) to push changes in your schema to the database
+- use [Prisma Client](/concepts/components/prisma-client) in your application to talk to the database server at PlanetScale
+
+## Differences to consider
+
+PlanetScale's branching model and design for scalability means that there are also a number of differences to consider. You should be aware of the following points when deciding to use PlanetScale with Prisma:
+
+- **Referential integrity.** To support scaling across multiple database servers, PlanetScale [does not allow the use of foreign key constraints](https://docs.planetscale.com/tutorials/operating-without-foreign-keys), which are normally used in relational databases to enforce relationships between data in different tables. To maintain these relationships in your data while using PlanetScale with Prisma, you will need to use Prisma's ability to [emulate referential integrity in the Prisma client](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-integrity#handling-the-referential-integrity-in-prisma). For more information, see [How to enable emulation of referential integrity](#how-to-enable-emulation-of-referential-integrity).
+
+- **Branching and deploy requests.** PlanetScale provides two types of database branches: **development branches**, which allow you to test out schema changes, and **production branches**, which are protected from direct schema changes. Production branches are highly available and include automated daily backups. Once you are happy with the changes you make on a development branch, you can create a deploy request, which you can then review before deploying the changes to production. To learn more, see [How to use branches and deploy requests](#how-to-use-branches-and-deploy-requests).
+
+- **Creating indexes on foreign keys.** In a standard MySQL database, if a table has a column with a foreign key constraint, an index is automatically created on that column. As PlanetScale does not support foreign keys, these indexes are not created which can lead to issues with queries not being well optimised. To avoid this, you can create indexes in Prisma. For more information, see [How to create indexes on foreign keys](#how-to-create-indexes-on-foreign-keys).
```

Comment:
This difference is different than the others. It is just a temporary thing that we will probably implement a solution for before `referentialIntegrity` leaves the preview feature stage.

Reply:
Any suggestions on how to include this information?

Reply:
"Creating indexes on foreign keys" could become a sub item of "Referential integrity", that I think would create the connection. And then add a "currently" to "these indexes are not created" and link to an issue we have about that: https://github.com/prisma/prisma/issues/10611 That would make clearer this might go away and be fixed.